wcfsetup/install/files/js/WCF.Attachment.js

```diff
@@ -285,20 +292,112 @@ if (COMPILER_TARGET_DEFAULT) {
 		 * @see        WCF.Upload._upload()
 		 */
 		_upload: function (event, file, blob) {
-			var $uploadID = undefined;
+			var _super = this._super.bind(this);
 			
-			if (this._validateLimit()) {
-				$uploadID = this._super(event, file, blob);
-			}
-			
-			if (this._fileUpload) {
-				// remove and re-create the upload button since the 'files' property
-				// of the input field is readonly thus it can't be reset
-				this._removeButton();
-				this._createButton();
-			}
-			
-			return $uploadID;
+			require([
+				'WoltLabSuite/Core/FileUtil',
+				'WoltLabSuite/Core/ImageResizer',
+				'WoltLabSuite/Core/Ajax/Status'
+			]).then((function (modules) {
+				var FileUtil = modules[0];
+				var ImageResizer = modules[1];
+				var AjaxStatus = modules[2];
+				
+				AjaxStatus.show();
+				
+				var $files = [];
+				
+				if (file) {
+					$files.push(file);
+				}
+				else if (blob) {
+					$files.push(FileUtil.blobToFile(blob, 'pasted-from-clipboard'));
+				}
+				else {
+					$files = this._fileUpload.prop('files');
+				}
+				
+				// Default action is to resolve with the unaltered list of files
+				var $promise = Promise.resolve($files);
+				
+				if (this._options.autoScale && this._options.autoScale.enable) {
+					var $maxSize = this._buttonSelector.data('maxSize');
+					
+					// Create an ImageResizer instance
+					var $resizer = new ImageResizer();
+					var $maxWidth = this._options.autoScale.maxWidth;
+					var $maxHeight = this._options.autoScale.maxHeight;
+					var $quality = this._options.autoScale.quality;
+					
+					if (this._options.autoScale.fileType !== 'keep') {
+						$resizer.setFileType(this._options.autoScale.fileType);
+					}
+					
+					// Resize the images in series.
+					// As our resizer is based on Pica it will use multiple workers per image if possible.
+					$promise = Array.prototype.reduce.call($files, (function (acc, file) {
+						return acc.then((function (arr) {
+							var $timeout = new Promise(function (resolve, reject) {
+								setTimeout(function () {
+									resolve(file);
+								}, 10e3); // Timeout per image
```

Comment:
`10000` is much more legible than `10e3`, you could even write it as `10 * 1000` if you're concerned about this. Also, the comment should highlight that the timeout is asynchronous for each image and thus runs in parallel.